src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx

```diff
@@ -108,18 +115,143 @@ export const BorrowInput = ({
     [onDepositAmountChange],
   )
 
+  const collateralAccountFilter = useMemo(
+    () => ({ accountId: collateralAccountId }),
+    [collateralAccountId],
+  )
+  const collateralAccountMetadata = useAppSelector(state =>
+    selectPortfolioAccountMetadataByAccountId(state, collateralAccountFilter),
+  )
+
+  const getBorrowFromAddress = useCallback(() => {
+    if (!(wallet && collateralAccountMetadata)) return null
+    return getThorchainFromAddress({
+      accountId: collateralAccountId,
+      assetId: collateralAssetId,
+      getPosition: getThorchainLendingPosition,
+      accountMetadata: collateralAccountMetadata,
+      wallet,
+    })
+  }, [collateralAccountId, collateralAccountMetadata, collateralAssetId, wallet])
+
+  useEffect(() => {
+    if (fromAddress) return
+    ;(async () => {
+      const _fromAddress = await getBorrowFromAddress()
+      if (!_fromAddress) return
+      setFromAddress(_fromAddress)
+    })()
+  }, [getBorrowFromAddress, fromAddress])
+
+  const {
+    data: estimatedFeesData,
+    isLoading: isEstimatedFeesDataLoading,
+    isSuccess: isEstimatedFeesDataSuccess,
+  } = useQuoteEstimatedFeesQuery({
+    collateralAssetId,
+    collateralAccountId,
+    borrowAccountId,
+    borrowAssetId: borrowAsset?.assetId ?? '',
+    depositAmountCryptoPrecision: cryptoDepositAmount ?? '0',
+  })
+
   const balanceFilter = useMemo(
     () => ({ assetId: collateralAssetId, accountId: collateralAccountId }),
     [collateralAssetId, collateralAccountId],
   )
-  const balance = useAppSelector(state =>
+  const balanceCryptoBaseUnit = useAppSelector(state =>
     selectPortfolioCryptoBalanceBaseUnitByFilter(state, balanceFilter),
   )
   const amountAvailableCryptoPrecision = useMemo(
-    () => bnOrZero(balance).div(bn(10).pow(collateralAsset?.precision ?? '0')),
-    [balance, collateralAsset?.precision],
+    () => bnOrZero(balanceCryptoBaseUnit).div(bn(10).pow(collateralAsset?.precision ?? '0')),
+    [balanceCryptoBaseUnit, collateralAsset?.precision],
+  )
+
+  const hasEnoughBalanceForTx = useMemo(
+    () =>
+      bnOrZero(cryptoDepositAmount)
+        .plus(
+          bnOrZero(estimatedFeesData?.txFeeCryptoBaseUnit).div(
+            bn(10).pow(collateralAsset?.precision ?? '0'),
+          ),
+        )
+        .lte(amountAvailableCryptoPrecision),
+    [
+      amountAvailableCryptoPrecision,
+      collateralAsset?.precision,
+      cryptoDepositAmount,
+      estimatedFeesData?.txFeeCryptoBaseUnit,
+    ],
+  )
+
+  const isSweepNeededArgs = useMemo(
+    () => ({
+      assetId: collateralAssetId,
+      address: fromAddress,
+      amountCryptoBaseUnit: bnOrZero(cryptoDepositAmount ?? '0')
```

Comment:
would be nice to leverage our `toBaseUnit` helper functions where applicable

Reply:
There's going to be a lot more in the PR above this one, happy to do a full cleanup of these there while at it!